internal/librarian/generate.go

```diff
@@ -212,23 +210,35 @@ func (r *generateRunner) runGenerateCommand(ctx context.Context, outputDir strin
 	return "", fmt.Errorf("library not found")
 }
 
+// runBuildCommand orchestrates the building of an API library using a containerized
+// environment.
+//
+// The `outputDir` parameter specifies the target directory where the built artifacts
+// should be placed.
 func (r *generateRunner) runBuildCommand(ctx context.Context, outputDir, libraryID string) error {
 	if !r.cfg.Build {
 		slog.Info("Build flag not specified, skipping")
 		return nil
 	}
-	if libraryID != "" {
-		slog.Info("Build requested in the context of refined generation; cleaning and copying code to the local language repo before building.")
-		// TODO(https://github.com/googleapis/librarian/issues/775)
-		if err := os.CopyFS(r.repo.Dir, os.DirFS(outputDir)); err != nil {
-			return err
-		}
-		if err := r.containerClient.Build(ctx, r.cfg, r.repo.Dir, libraryID); err != nil {
-			return err
-		}
+	if libraryID == "" {
+		slog.Warn("Cannot perform build, missing library ID")
+		return nil
 	}
-	slog.Warn("Cannot perform build, missing library ID")
-	return nil
+
+	buildRequest := &docker.BuildRequest{
+		Cfg:       r.cfg,
+		State:     r.state,
+		LibraryID: libraryID,
+		RepoDir:   r.repo.Dir,
+	}
+
+	slog.Info("Build requested in the context of refined generation; cleaning and copying code to the local language repo before building.")
+	// TODO(https://github.com/googleapis/librarian/issues/775)
+	if err := os.CopyFS(r.repo.Dir, os.DirFS(outputDir)); err != nil {
+		return err
+	}
+
+	return r.containerClient.Build(ctx, buildRequest)
```

Comment:
I missed this in the previous "generate" PR. But should we cleanup the json file after execution? Otherwise, we will end up with a lot of old generate/build request files in .librarian/ dir.

Reply:
How about creating an clean up issue so that I can remove this file in a follow up PR?